cmd/settings/main.go

```diff
@@ -37,16 +38,30 @@ func main() {
 	flag.Parse()
 
 	cfg.Version = uint16(*versionU64)
+
+	// check that interval is valid
+	validInterval, _ := regexp.MatchString("[0-9]", string([]rune(cfg.BlockProductionInterval)[0]))
+	if !validInterval {
+		log.Fatalf("Interval input did not start with digit")
+	}
+
+	validInterval, _ = regexp.MatchString("[a-s]", string([]rune(cfg.BlockProductionInterval)[len(cfg.BlockProductionInterval)-1]))
+	if !validInterval {
+		log.Fatalf("Interval input did not end with letter")
+	}
 	//check block production interval suffix
 	hasSuf := false
+
 	for _, s := range [7]string{"ns", "us", "µs", "ms", "s", "m", "h"} {
 		if strings.HasSuffix(cfg.BlockProductionInterval, s) {
 			hasSuf = true
```

Comment:
Instead of the `validInterval` above, can we instead leverage this function: https://golang.org/pkg/strings/#TrimSuffix

We can use it to validate that everything except for the suffix is a digit. The problem with `validInterval` is it would probably accept `1yns`, but I'm not sure cause there's no test for it.

Reply:
I ended up using TrimLeftFunc but I believe you shall enjoy and the validation should be more robust now. I made a card to clean up this file and add testing

Reply:
Sir I'm running `go run main.go` and the interval is not changing in the `config.json` file.

Let's make this validation a testable function in the `validation` package.